packages/smooth_app/lib/cards/product_cards/question_card.dart

```diff
@@ -377,14 +381,16 @@ Future<void> saveAnswer({
   required String? insightId,
   required InsightAnnotation insightAnnotation,
 }) async {
-  // TODO(jasmeet): Send answer to the Backend.
-  // TODO(jasmeet): Fix for iOS. https://pub.dev/packages/device_info
-  // DeviceInfoPlugin deviceInfo = DeviceInfoPlugin();
-  // AndroidDeviceInfo androidInfo = await deviceInfo.androidInfo;
-  //
-  // final Status status = await OpenFoodAPIClient.postInsightAnnotation(
-  //   insightId,
-  //   insightAnnotation,
-  //   deviceId: androidInfo.id,
-  // );
+  final DeviceInfoPlugin deviceInfoPlugin = DeviceInfoPlugin();
+  String? deviceId;
+  if (Platform.isAndroid) {
+    deviceId = (await deviceInfoPlugin.androidInfo).androidId;
+  } else if (Platform.isIOS) {
+    deviceId = (await deviceInfoPlugin.iosInfo).identifierForVendor;
+  }
```

Comment:
why exception ? The idea of the deviceId is that for users that are not signed in, the backend stores the answer with the device Id of the user, later we can even have badges for users that aren't signed in, as long as they use the same client. 

In case deviceId is not found we will just send null. Lets not throw an exception, because the user still answered the question, we should leverage the answer.

Reply:
You test `isAndroid` and `isIOS`. If it's not one of them, it's not normal, is it?
Beyond that, it's ok to leave the `deviceId` to `null`.

Reply:
yeah, not normal for sure, but I still don't think we get anything from throwing an exception and halting the flow just in case it happens.

Reply:
The point is not to halt "normal" users, that will use android or ios anyway. The point is rather to:
* double-check "abnormal" uses
* if ever one day we develop beyond android and ios, the app will rapidly crash here and we'll be able to fix directly the code for the additional platform - if we don't do that everything will look ok but with a systematic `null` deviceid for those new platforms.

That's not a critical point, though.

Reply:
ok, but I also don't want to halt "abnormal" users either. I'm guessing it would happen if we launch this app to non ios/android users, but if we end up in this situation we should still let the app function seamlessly for these platform users. I added a `debugPrint` statement, does it generate sentry logs ? If it does then we would be notified of this code that wasn't updated while launching to another platform. Is that a good compromise ?

Reply:
> ok, but I also don't want to halt "abnormal" users either.

I do. It's much easier to detect bugs in an app that crashes than in an app that silently does something different from what was expected.
And in this particular case it's a bit like when dart says: "oh by the way you did not include all the `enum`s values in your `switch`, that doesn't look good`.
`debugPrint` does just what it stand for: writing in the printed logs (not related to Sentry).

Again, it's not a critical point.

Reply:
> > ok, but I also don't want to halt "abnormal" users either.
> 
> I do. It's much easier to detect bugs in an app that crashes than in an app that silently does something different from what was expected. And in this particular case it's a bit like when dart says: "oh by the way you did not include all the `enum`s values in your `switch`, that doesn't look good`. `debugPrint` does just what it stand for: writing in the printed logs (not related to Sentry).

To be honest I don't think it's critical enough to throw exception, that's why I'm against it. The worst case is that the user on this new platform doesn't get contribution point. That's way better than showing an error and discouraging contribution.

How can I record an exception in Sentry without halting the flow, is it possible?

> 
> Again, it's not a critical point.

agreed, but just wanted to talk through.